src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala

```diff
@@ -0,0 +1,113 @@
+package mesosphere.marathon
+package api.akkahttp.v2
+
+import akka.http.scaladsl.model.Uri.Path
+import akka.http.scaladsl.server.Route
+import mesosphere.marathon.api.akkahttp.Controller
+import mesosphere.marathon.api.akkahttp.Directives.extractInstanceId
+import mesosphere.marathon.api.akkahttp.PathMatchers.{ ExistingRunSpecId, PodsPathIdLike }
+import mesosphere.marathon.core.group.GroupManager
+import mesosphere.marathon.core.instance.Instance
+import mesosphere.marathon.plugin.auth.Authenticator
+import mesosphere.marathon.state.PathId
+import akka.http.scaladsl.server.PathMatchers
+
+class PodsController(
+    val groupManager: GroupManager)(
+    implicit
+    val authenticator: Authenticator) extends Controller {
+
+  import mesosphere.marathon.api.akkahttp.Directives._
+
+  def capability(): Route =
+    authenticated.apply { implicit identity =>
+      complete("")
+    }
+
+  def create(): Route = ???
+
+  def update(podId: PathId): Route = ???
+
+  def findAll(): Route = ???
+
+  def find(podId: PathId): Route = ???
+
+  def remove(podId: PathId): Route = ???
+
+  def status(podId: PathId): Route = ???
+
+  def versions(podId: PathId): Route = ???
+
+  def version(podId: PathId, v: String): Route = ???
+
+  def allStatus(): Route = ???
+
+  def killInstance(instanceId: Instance.Id): Route = ???
+
+  def killInstances(podId: PathId): Route = ???
+
+  // format: OFF
+  override val route: Route =
+    head {
+      capability()
+    } ~
+    get {
+      pathEnd {
+        findAll()
+      } ~
+      path("::status" ~ PathEnd) {
+        allStatus()
+      } ~
+      path(PodsPathIdLike ~ PathEnd) { runSpecId: String =>
+        assumeValid(validatePathId(runSpecId)) {
+          find(PathId(runSpecId))
+        }
+      } ~
+      path(PodsPathIdLike ~ "::status" ~ PathEnd) { runSpecId: String =>
+        assumeValid(validatePathId(runSpecId)) {
```

Comment:
to make it even more readable here I would probably go towards direction of having a tailor-made directive for this type called assumeValidPathId but it's up to a discussion. Your approach is better if we want to validate many different entities... But I am not sure whether that is the case.